pilot/pkg/networking/core/v1alpha3/networkfilter.go

```diff
@@ -169,25 +173,34 @@ func buildOutboundNetworkFiltersWithWeightedClusters(env *model.Environment, nod
 		}
 	}
 
+	// TODO: Need to handle multiple cluster names for Redis
+	clusterName := clusterSpecifier.WeightedClusters.Clusters[0].Name
 	tcpFilter := setAccessLogAndBuildTCPFilter(env, node, proxyConfig)
-	return buildNetworkFiltersStack(node, port, tcpFilter, statPrefix)
+	return buildNetworkFiltersStack(node, port, tcpFilter, statPrefix, clusterName)
 }
 
 // buildNetworkFiltersStack builds a slice of network filters based on
 // the protocol in use and the given TCP filter instance.
-func buildNetworkFiltersStack(node *model.Proxy, port *model.Port, tcpFilter *listener.Filter, statPrefix string) []listener.Filter {
+func buildNetworkFiltersStack(node *model.Proxy, port *model.Port, tcpFilter *listener.Filter, statPrefix string, clusterName string) []listener.Filter {
 	filterstack := make([]listener.Filter, 0)
 	switch port.Protocol {
 	case model.ProtocolMongo:
-		filterstack = append(filterstack, buildMongoFilter(statPrefix, util.IsProxyVersionGE11(node)), *tcpFilter)
+		filterstack = append(filterstack, buildMongoFilter(statPrefix, util.IsXDSMarshalingToAnyEnabled(node)), *tcpFilter)
+	case model.ProtocolRedis:
+		if util.IsProxyVersionGE11(node) && pilot.EnableRedisFilter() {
+			filterstack = append(filterstack, buildRedisFilter(statPrefix, clusterName, util.IsXDSMarshalingToAnyEnabled(node)))
```

Comment:
Mind adding a comment as to why we don't include the TCP Proxy here?

Reply:
done